CRM/Financial/BAO/PaymentProcessor.php

```diff
@@ -273,7 +273,7 @@ public static function defaultComparison($processor1, $processor2) {
    */
   public static function getAllPaymentProcessors($mode = 'all', $reset = FALSE, $isCurrentDomainOnly = TRUE) {
 
-    $cacheKey = 'CRM_Financial_BAO_Payment_Processor_' . $mode . '_' . CRM_Core_Config::domainID();
+    $cacheKey = 'CRM_Financial_BAO_Payment_Processor_' . $mode . '_' . $isCurrentDomainOnly . '_' . CRM_Core_Config::domainID();
```

Comment:
I guess you could drop off domain if not current domain only - but I think any benefit from that is negligible & on balance this feels slightly safer